Check for ext protocol support

sewnie opened the pull request:
With the toplevel protocol being supported, no NULL check was put in place for it if a compositor does not support it. Here, the check was rewritten to check for the screencast type's required protocol.

Fixes #337

kennylevinsen reviewed (commented):
How did you trigger this crash?

`state->screencast_source_types` only includes `WINDOW` if `ext_foreign_toplevel_iamge_capture_source_manager` is set, so I'm surprised if a client is asking for it.

I suspect it would be cleaner to handle this in `setup_target` instead anyway, to ensure we don't show options to the user that we shouldn't have:

```c
type_mask &= ctx->state->screencast_source_types;
if (type_mask == 0) {
    logprint(ERROR, "wlroots: No supported targets specified");
    return;
}
```

sewnie committed:
Check for ext protocol support

With the toplevel protocol being supported, no NULL check was put
in place for it if a compositor does not support it. Here, the check
was rewritten to check for the screencats type's required protocol.
